controllers/update_backup_agents.go

```diff
@@ -84,6 +85,7 @@ func (u updateBackupAgents) reconcile(ctx context.Context, r *FoundationDBBackup
 	}
 
 	if !needCreation && deployment == nil {
+		log.Info("Deleting deployment", "name", existingDeployment.Name)
```

Comment:
That should be a debug statement not an info statement. I would prefer if we also log the corresponding namespace and cluster name.